CHANGELOG.md

```diff
@@ -21,6 +21,8 @@
 - [Custom Types](https://mozilla.github.io/uniffi-rs/proc_macro/index.html#the-unifficustomtype-derive) are now supported for proc-macros, including a very
   low-friction way of exposing types implementing the new-type idiom.
 - Proc-macros: Added support for ByRef arguments
+- Swift async functions/methods now always throw, since they can throw CancellationError when the parent
+  task is cancelled.
```

Comment:
I have no clue :)  The only alternative I could see was a runtime error, which can't be a general solution.  Maybe users could configure which behavior they wanted?

Reply:
Let's ask good Swift developers, like @stefanceriu or @pixlwave?

Reply:
I'm not sure I get the full context here but that code looks just fine 🤷‍♂️

Reply:
It's about [this](https://github.com/mozilla/uniffi-rs/pull/1742/files#diff-c96520ae7c5e213facf72cbbad8297cdc353c894f23f4aae30dc298463471c9a) diff where every `await` is now prefixed with `try!` or `try?`.

Reply:
When you're dealing with throwing functions you need to mark them with some variation of `try`. When you're in the context of a Task:
* simple `try` will rethrow the error up to the task level, where it will transform into a `Task<(), Error>`
* `try?` will make the result an optional which you will need to handle locally. The parent task will become `Task<()?, Never>`
* and `try!` will make the runtime crash with a `Fatal error: 'try!' expression unexpectedly raised an error:`. The parent task will become `Task<(), Never>` (notice the optional)

I'm not sure how that testing jig is setup and how fatal errors are handled but it seems that tasks that are expected to throw are correctly marked with `try?`. For the rest it might be simpler to just use `try`, throw errors up the task level and expect the `counter` to be wrong and fail the test.

Hope this answers your question

Reply:
I think it's about the ergonomics for consumers.  We probably don't want to require them to add `try!` to all of their async function calls, because they could be cancelled in principle even if the application never actually cancels them.

I think you suggested configuration as a solution to this issue and that seems like it strikes a good balance.  By default, generated async functions will not check for cancellation and will only throw if the underlying Rust function returns a `Result`.  Swift users can opt-in to cancellation checks in the `uniffi.toml` file, which means they will always need to use one of the `try` variations when calling the function.